templates/addons/cluster-autoscaler/cluster-autoscaler.yaml

```diff
@@ -16,7 +16,7 @@ spec:
         app: ${CLUSTER_NAME}-cluster-autoscaler
     spec:
       containers:
-        - image: registry.k8s.io/autoscaling/cluster-autoscaler:${CLUSTER_AUTOSCALER_VERSION:=v9.37.0}
+        - image: registry.k8s.io/autoscaling/cluster-autoscaler:${CLUSTER_AUTOSCALER_VERSION:=v1.30.2}
```

Comment:
I think if we stick to use Bash (or some other shell language with Bash-like [parameter expansion](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html)), we can use parameter expansion here:

```suggestion
        - image: registry.k8s.io/autoscaling/cluster-autoscaler:${KUBERNETES_VERSION%.*}.0
```

This snippet matches the shortest trailing pattern of `${KUBERNETES_VERSION}` up until the first `.` character, i.e. gets the major and minor Kubernetes version. Then it sets the Cluster Autoscaler version to the very first release (`vX.X.0`) (i.e. ignores any patch versions) of that Kubernetes version.

Testing this out, I get:
```sh
$ clusterctl generate cluster test --kubernetes-version v1.29.7+k3s1 --infrastructure local-linode:v0.0.0 --flavor k3s-cluster-autoscaler | yq 'select(.kind == "Deployment").spec.template.spec.containers[].image'
registry.k8s.io/autoscaling/cluster-autoscaler:v1.29.0

$ clusterctl generate cluster test --kubernetes-version v1.30.1+rke2r1 --infrastructure local-linode:v0.0.0 --flavor rke2-cluster-autoscaler | yq 'select(.kind == "Deployment").spec.template.spec.containers[].image'
registry.k8s.io/autoscaling/cluster-autoscaler:v1.30.0
```